### ISSUE TITLE
renameTag.php HTTP 500 error

### ISSUE THREAD

**Issue by HuidaeCho** [nextcloud/bookmarks#117]
When I try to rename tags, I'm getting an HTTP 500 error. I found a couple issues in the renameTag routine in apps/bookmarks/lib/bookmarks.php. Hope this patch helps.

``` diff
--- apps/bookmarks/lib/bookmarks.php.orig  2014-12-13 22:30:27.050092385 -0500
+++ apps/bookmarks/lib/bookmarks.php    2014-12-30 01:23:36.705682456 -0500
@@ -192,11 +192,12 @@
                        $query = OCP\DB::prepare("
                        DELETE FROM `*PREFIX*bookmarks_tags` as `tgs` WHERE `tgs`.`tag` = ?
                        AND exists( SELECT `id` FROM `*PREFIX*bookmarks` WHERE `user_id` = ? AND `tgs`.`bookmark_id` = `id`)
-                       AND exists( SELECT `t`.`tag` FROM `*PREFIX*bookmarks_tags` `t` where `t`.`tag` = ? AND `tgs`.`bookmark_id` = `t`.`bookmark_id`");
+                       AND exists( SELECT `t`.`tag` FROM `*PREFIX*bookmarks_tags` `t` where `t`.`tag` = ? AND `tgs`.`bookmark_id` = `t`.`bookmark_id`)");

                        $params=array(
                                $new,
                                $user_id,
+                               $new,
                        );

                        $result = $query->execute($params);
@@ -214,7 +215,6 @@
                                $new,
                                $old,
                                $user_id,
-                               $old,
                        );

                        $result = $query->execute($params);
```

**Comment by Chouchen** [comment]
Hi,

can you do a PR with it?

**Comment by africanforest** [comment]
When i try to rename a tag, i'm getting no error, but renaming simply doesn't work. The tag always keeps its old name.
(OC 8.1)

**Comment by essig** [comment]
For renaming tags, the query has to be changed (see code below)
- "USING" has to be used for aliases
- "JOIN" has to be used for SELECT statement on same table within DELETE statement
```
$query = $db->prepareQuery("
	DELETE FROM `tgs` USING `*PREFIX*bookmarks_tags` as `tgs` 
	JOIN `*PREFIX*bookmarks_tags` as `t`
	ON `tgs`.`bookmark_id` = `t`.`bookmark_id`
	WHERE `tgs`.`tag` = ?
	AND `t`.`tag` = ?
	AND exists( SELECT `b`.`id` FROM `*PREFIX*bookmarks` `b` WHERE `b`.`user_id` = ?
	AND `tgs`.`bookmark_id` = `b`.`id`)");
```

**Comment by marcelklehr** [comment]
I think this was fixed by #332 and #337. Feel free to reopen / make some noise if you disagree.
